src/System.Net.Http/src/System/Net/Http/ByteArrayHelpers.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System
+{
+    internal static class ByteArrayHelpers
+    {
+        internal static bool EqualsOrdinalAsciiIgnoreCase(string left, ReadOnlySpan<byte> right)
+        {
+            Debug.Assert(left != null, "Expected non-null string");
+
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint charA = left[i];
+                uint charB = right[i];
+
+                unchecked
+                {
+                    // We're only interested in ASCII characters here.
+                    if ((charA - 'a') <= ('z' - 'a'))
+                        charA -= ('a' - 'A');
+                    if ((charB - 'a') <= ('z' - 'a'))
+                        charB -= ('a' - 'A');
+                }
+
+                if (charA != charB)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        internal static string GetStringFromByteSpan(ReadOnlySpan<byte> bytes)
+        {
+            string s;
+            unsafe
+            {
+                fixed (byte* p = &bytes.DangerousGetPinnableReference())
+                {
+                    s = Encoding.ASCII.GetString(p, bytes.Length);
+                }
+            }
+
+            return s;
+        }
```

Comment:
If the code calling this is only used from a .NET Core build, you don't need this helper and can instead just do:
```
Encoding.ASCII.GetString(bytes);
```
once https://github.com/dotnet/corefx/pull/23049 is merged, due to this new member:
https://github.com/dotnet/corefx/pull/23049/files#diff-b3d39e42f75b9efe89d21eb385ed61f6R7350

Otherwise, nit, you could consolidate this a little bit, e.g.
```C#
internal static unsafe string GetStringFromByteSpan(ReadOnlySpan<byte> bytes)
{
    fixed (byte* p = &bytes.DangerousGetPinnableReference())
    {
        return Encoding.ASCII.GetString(p, bytes.Length);
    }
}
```